extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -141,185 +76,217 @@ export abstract class AzureAuth implements vscode.Disposable {
 	}
 
-	public async clearCredentials(account: azdata.AccountKey): Promise<void> {
-		try {
-			return this.deleteAccountCache(account);
-		} catch (ex) {
-			const msg = localize('azure.cacheErrrorRemove', "Error when removing your account from the cache.");
-			vscode.window.showErrorMessage(msg);
-			Logger.error('Error when removing tokens.', ex);
+	protected async getToken(tenant: Tenant, resource: Resource, postData: AuthorizationCodePostData | TokenPostData | RefreshTokenPostData): Promise<OAuthTokenResponse> {
+		const tokenUrl = `${this.loginEndpointUrl}${tenant.id}/oauth2/token`;
+		const response = await this.makePostRequest(tokenUrl, postData);
+
+		if (response.data.error === 'interaction_required') {
+			return this.handleInteractionRequired(tenant, resource);
 		}
-	}
 
-	protected toBase64UrlEncoding(base64string: string): string {
-		return base64string.replace(/=/g, '').replace(/\+/g, '-').replace(/\//g, '_'); // Need to use base64url encoding
+		const accessTokenString = response.data?.access_token;
+		const refreshTokenString = response.data?.refresh_token;
+		const expiresOnString = response.data.expires_on;
+
+		return this.getTokenHelper(tenant, resource, accessTokenString, refreshTokenString, expiresOnString);
 	}
 
-	protected async makePostRequest(uri: string, postData: { [key: string]: string }, validateStatus = false) {
-		try {
-			const config: AxiosRequestConfig = {
-				headers: {
-					'Content-Type': 'application/x-www-form-urlencoded'
-				}
-			};
+	protected async getTokenHelper(tenant: Tenant, resource: Resource, accessTokenString: string, refreshTokenString: string, expiresOnString: string): Promise<OAuthTokenResponse> {
+		if (!accessTokenString) {
+			const msg = localize('azure.accessTokenEmpty', 'No access token returned from Microsoft OAuth');
+			throw new AzureAuthError(msg, 'Access token was empty', undefined);
+		}
 
-			if (validateStatus) {
-				config.validateStatus = () => true;
-			}
+		const tokenClaims: TokenClaims = this.getTokenClaims(accessTokenString);
 
-			return await axios.post(uri, qs.stringify(postData), config);
-		} catch (ex) {
-			Logger.log('Unexpected error making Azure auth request', 'azureCore.postRequest', JSON.stringify(ex?.response?.data, undefined, 2));
-			throw ex;
+		const userKey = tokenClaims.sub ?? tokenClaims.oid;
+
+		if (!userKey) {
+			const msg = localize('azure.noUniqueIdentifier', "The user had no unique identifier within AAD");
+			throw new AzureAuthError(msg, 'No unique identifier', undefined);
 		}
-	}
 
-	protected async makeGetRequest(token: string, uri: string): Promise<AxiosResponse<any>> {
-		try {
-			const config = {
-				headers: {
-					Authorization: `Bearer ${token}`,
-					'Content-Type': 'application/json',
-				},
-			};
+		const accessToken: AccessToken = {
+			token: accessTokenString,
+			key: userKey
+		};
+		let refreshToken: RefreshToken;
+		const expiresOn: string = expiresOnString;
```

Comment:
Why not just use expiresOnString directly?

Reply:
I thought it was cleaner like then when writing it but looking at it again I don't think it is haha.